### ISSUE TITLE
Issues parsing tailf-common yang module

### ISSUE THREAD

**Issue by demonicblue** [corenova/yang-js#16]
I'm having issues parsing the yang files that comes with ConfD, specifically tailf-common. The error reads
```
{ [ExpressionError: [constructor] invalid YANG syntax detected]
  name: 'ExpressionError',
  context: 'tring { pattern "S(\\d+G)?(\\d+M)?(\\d+K)?(\\d+B)?' }
```
and the corresponding code is
```
  typedef size {
    type string {
      pattern
        "S(\d+G)?(\d+M)?(\d+K)?(\d+B)?";
    }
    description
      "A value that represents a number of bytes.  An example could be
       S1G8M7K956B; meaning 1GB + 8MB + 7KB + 956B = 1082138556 bytes.
       The value must start with an S.  Any byte magnifier can be left
       out, e.g. S1K1B equals 1025 bytes.  The order is significant
       though, i.e. S1B56G is not a valid byte size.

       In ConfD, a 'size' value is represented as an uint64.";
  }
```
However, parsing the typedef out of the context of the tailf-common module renders no errors.

Any ideas on how to debug?

**Comment by saintkepha** [comment]
That particular error is during pure syntax parsing by the "yang-parser" module.  One option you can try is to use that module directly and see what error you get:

```
var parser = require('yang-parser');
parser.parse( schema );
```

This should give you the error dump AS-IS. 

Otherwise, send me the .yang file in question and I can try to reproduce to see what the issue may be.  I'll also check on whether syntax error content extraction logic in yang-js is working properly.

**Comment by saintkepha** [comment]
The current issue is that `yang-parser` module is not able to parse:
```
pattern "S(\d+G)?(\d+M)?(\d+K)?(\d+B)?";
```
When you change the double quote into single quote then the parsing issue is bypassed.
Will need to look into how to address this at some point.

**Comment by saintkepha** [comment]
FYI - while testing issue #20 I had to also apply similar changes to the `tailf-meta-extensions.yang` regarding the single quotes for the `pattern` statements.

**Comment by saintkepha** [comment]
Proper fix for handling escape characters in double quoted string resolved with 0.16.23
